lib/cc/analyzer/engine.rb

```diff
@@ -33,7 +33,7 @@ def run(stdout_io, container_listener)
 
         container.on_output("\0") do |output|
           unless output_filter.filter?(output)
-            stdout_io.write(output)
+            container.stop unless stdout_io.write(output)
```

Comment:
> I don't usually advocate using or but given that it's for control flow and not logic, I think it's a good use case.

FWIW, I love `and` / `or` for control flow. That's what they're there for; it's why the lower precedence.

```rb
foo_bar or raise "x"

x = get_it and go(x)
```

We seem to avoid them here, so I've been abiding but I'd be happy to have a discussion about bringing them back into our toolbox.